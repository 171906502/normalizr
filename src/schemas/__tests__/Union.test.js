/* eslint-env jest */
import { normalize, schema } from '../../';

describe(schema.Union.name, () => {
  it('normalizes an array of multiple entities using string schemaAttribute', () => {
    const user = new schema.Entity('users');
    const group = new schema.Entity('groups');
    const union = new schema.Union({
      users: user,
      groups: group
    }, 'type');

    expect(normalize([ { id: 1, type: 'users' }, { id: 2, type: 'groups' } ], union)).toMatchSnapshot();
  });

  it('normalizes an array of multiple entities using a function to infer the schemaAttribute', () => {
    const user = new schema.Entity('users');
    const group = new schema.Entity('groups');
    const union = new schema.Union({
      users: user,
      groups: group
    }, (input) => { return input.username ? 'users' : 'groups'; });

    expect(normalize([ { id: 1, username: 'Janey' }, { id: 2, groupname: 'People' } ], union)).toMatchSnapshot();
  });
});
