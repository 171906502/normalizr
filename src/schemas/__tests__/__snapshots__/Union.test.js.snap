exports[`UnionSchema normalizes an array of multiple entities using a function to infer the schemaAttribute 1`] = `
Object {
  "entities": Object {
    "groups": Object {
      "2": Object {
        "groupname": "People",
        "id": 2,
      },
    },
    "users": Object {
      "1": Object {
        "id": 1,
        "username": "Janey",
      },
    },
  },
  "result": Array [
    Object {
      "id": 1,
      "schema": "users",
    },
    Object {
      "id": 2,
      "schema": "groups",
    },
  ],
}
`;

exports[`UnionSchema normalizes an array of multiple entities using string schemaAttribute 1`] = `
Object {
  "entities": Object {
    "groups": Object {
      "2": Object {
        "id": 2,
        "type": "group",
      },
    },
    "users": Object {
      "1": Object {
        "id": 1,
        "type": "user",
      },
    },
  },
  "result": Array [
    Object {
      "id": 1,
      "schema": "users",
    },
    Object {
      "id": 2,
      "schema": "groups",
    },
  ],
}
`;
